src/main/java/org/primefaces/component/datatable/DataTable.java

```diff
@@ -1529,11 +1543,139 @@ public boolean isSortingCurrentlyActive() {
+
+            updateFilterByWithUserFilterBy(filterBy, filterByTmp, filtered);
+        }
+
+        setDefaultFilter(filtered.get());
+
+        return filterBy;
+    }
+
+    protected void populateFilterBy(FacesContext context, char separator, UIComponent root, Map<String, FilterMeta> filterBy, AtomicBoolean defaultFilter) {
+        for (int i = 0; i < root.getChildCount(); i++) {
+            UIComponent child = root.getChildren().get(i);
+            if (child.isRendered()) {
+                if (child instanceof Columns) {
+                    Columns columns = (Columns) child;
+                    String uiColumnsClientId = columns.getClientId(context);
+
+                    for (int j = 0; j < columns.getRowCount(); j++) {
+                        DynamicColumn dynaColumn = new DynamicColumn(j, columns);
+                        dynaColumn.setColumnKey(uiColumnsClientId + separator + j);
+                        FilterMeta f = FilterMeta.of(getFacesContext(), getVar(), dynaColumn);
+                        if (f != null) {
+                            filterBy.put(f.getColumnKey(), f);
+                            defaultFilter.set(defaultFilter.get() || f.isActive());
+                        }
+                    }
+                }
+                else if (child instanceof UIColumn) {
+                    UIColumn column = (UIColumn) child;
+                    FilterMeta f = FilterMeta.of(getFacesContext(), getVar(), column);
+                    if (f != null) {
+                        filterBy.put(f.getColumnKey(), f);
+                        defaultFilter.set(defaultFilter.get() || f.isActive());
+                    }
+                }
+                else if (child instanceof Row) {
+                    populateFilterBy(context, separator, child, filterBy, defaultFilter);
+                }
+                else if (child instanceof ColumnGroup) {
+                    populateFilterBy(context, separator, child, filterBy, defaultFilter);
+                }
+            }
+        }
+    }
+
+    protected void updateFilterByWithTableState(Map<String, FilterMeta> tsSortBy) {
+        if (tsSortBy != null) {
+            boolean defaultFilter = isDefaultFilter();
+            for (Map.Entry<String, FilterMeta> entry : tsSortBy.entrySet()) {
+                FilterMeta intlSortBy = getFilterByAsMap().get(entry.getKey());
+                if (intlSortBy != null) {
+                    FilterMeta tsSortMeta = entry.getValue();
+                    intlSortBy.setFilterValue(tsSortMeta.getFilterValue());
+                    defaultFilter |= intlSortBy.isActive();
+                }
+            }
+
+            setDefaultFilter(defaultFilter);
+        }
+    }
+
+    protected void updateFilterByWithUserFilterBy(Map<String, FilterMeta> intlFilterBy, Collection<FilterMeta> usrFilterBy, AtomicBoolean filtered) {
+        for (FilterMeta userFM : usrFilterBy) {
+            FilterMeta intlSM = intlFilterBy.values().stream()
```

Comment:
I don't understand the variable name `intlSM` I can't figure out what that is short for should it be `intlFM` for FilterMeta?

Reply:
Yes indeed, I copy/paste from previous PR